crates/xtask/src/changelog.rs

```diff
@@ -154,7 +165,28 @@ impl Changelog {
         );
         ensure!(last_release.contents.is_empty(), "The last release must contain no changes");
 
-        Ok(Changelog { releases, skip_counter })
+        let changelog = Changelog { releases, skip_counter };
+        let output_string = format!("{changelog}");
+
+        ensure!(
+            output_string == contents,
+            "Input string does not equal rendered changelog.\n\nIn: {contents}\n\nOut: {contents}"
+        );
+
+        Ok(changelog)
+    }
+}
+
+impl Display for Changelog {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let releases_string = self.releases.iter().map(|release| format!("{release}")).join("\n\n");
```

Comment:
This is not efficient to build a string for each version and then join them. We want to output directly to the formatter.
```suggestion
        for release in &self.releases {
            writeln!(f, "{release}\n")?;
        }
```

Also this permits getting rid of `itertools` which is usually not useful.